client/src/components/IllnessStatusForm.vue

```diff
@@ -0,0 +1,204 @@
+          />
+        </a-form-item>
+      </a-col>
+      <a-col :md="12">
+        <a-form-item label="Meldedatum">
+          <DateInput
+            v-decorator="[
+              'dateOfReporting',
+              {
+                rules: [
+                  {
+                    required: true,
+                    message: 'Bitte Meldedatum wählen',
+                  },
+                ],
+                initialValue: initialDateOfReporting,
+              },
+            ]"
+          />
+        </a-form-item>
+      </a-col>
+    </a-row>
+
+    <a-divider />
+
+    <h4>Hospitalisierung</h4>
+    <a-row :gutter="12">
+      <a-col :md="12">
+        <a-form-item>
+          <a-checkbox v-model="patientHospitalized" v-decorator="['patientHospitalized']">
+            Patient/in ist hospitalisiert
+          </a-checkbox>
+        </a-form-item>
+      </a-col>
+      <a-col :md="12">
+        <a-form-item label="Datum der Hospitalisierung">
+          <DateInput
+            v-decorator="[
+              'dateOfHospitalization',
+              {
+                rules: [
+                  {
+                    required: patientHospitalized,
+                    message: 'Bitte Datum wählen',
+                  },
+                ],
+                initialValue: initialDateOfHospitalization,
+              },
+            ]"
+            :disabled="!patientHospitalized"
+            label="Seit"
+            style="flex: 0 1 400px;"
+          />
+        </a-form-item>
+        <a-form-item>
+          <a-checkbox
+            v-decorator="['onIntensiveCareUnit']"
+            :disabled="!patientHospitalized"
+          >
+            Auf der Intensivstation
+          </a-checkbox>
+        </a-form-item>
+      </a-col>
+    </a-row>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import moment, { Moment } from 'moment'
+import { EventTypeItem, eventTypes } from '@/models/event-types'
+import DateInput from '@/components/DateInput.vue'
+
+/**
+ * Autocomplete for Patients
+ *
+ * props:
+ * - validation: put in v-decorator object
+ */
+
+export interface State {
+  patientHospitalized: boolean
+  today: Moment
+  EVENT_TYPES: EventTypeItem[]
+  initialDateOfIllness: Moment | undefined
+  initialDateOfReporting: Moment | undefined
+  initialDateOfHospitalization: Moment | undefined
+}
+
+export default Vue.extend({
+  name: 'IllnessStatusForm',
+  props: ['form', 'patient'],
+  components: {
+    DateInput,
+  },
+  created() {
+    if (this.patient) {
+      if (this.patient.dateOfHospitalization) {
+        this.patientHospitalized = true
+        this.initialDateOfHospitalization = moment(this.patient.dateOfHospitalization)
+      } else {
+        this.initialDateOfHospitalization = this.today
+      }
+      if (this.patient.dateOfReporting) {
+        this.initialDateOfReporting = moment(this.patient.dateOfReporting)
+      } else {
+        this.initialDateOfReporting = this.today
+      }
```

Comment:
Auch hier wird das mit einem ternary operator viel übersichtlicher.

```
this.initialDateOfReporting = this.patient.dateOfReporting ? moment(this.patent.dateOfReporting) : moment()
```

Reply:
Generell würde ich auch lieber `moment()` statt `this.today` verwenden, damit immer eine neue moment Instanz zugewiesen wird.

Reply:
done